### DISCUSSION TITLE
config-validator.js throws Configuration Errors for sub keys under cacheTtlOverride

### DISCUSSION THREAD

**Discussion by lyonlai** [renovatebot/renovate#26977] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:37.159.1-slim

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

We set the preset cacheTtlOverride to an hour. However when we use `config-validator.js` to protect our pipeline. It throws errors `Invalid configuration option: cacheTtlOverride.preset`. 

```js
module.exports = {
 // ...
  cacheTtlOverride: {
      preset: 60,
  },
  //...
}
```

By looking into the `validateConfig` function in `lib/config/validation.js`, I that `cacheTtlOverride` is being validated recursively. Should we just ignore the recursive check for this key? Or there are existing validation logic around that's been missed?

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

 INFO: Validating config.js
ERROR: Found errors in configuration
       "file": "config.js",
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Invalid configuration option: cacheTtlOverride.preset"
         }
       ]
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by RahulGautamSingh** [comment, marked as answer]
The old validation logic is a bit broken atm for file config validation. The issue you are facing will get fixed in this PR: https://github.com/renovatebot/renovate/pull/25218

**Comment by lyonlai** [comment]
Cool. Thanks for that. expecting it to be merged. @RahulGautamSingh
